_std/macros/items.dm

```diff
@@ -14,3 +14,6 @@
 
 /// Returns true if the given x is a grab (obj/item/grab)
 #define isgrab(x) (istype(x, /obj/item/grab/))
+
+/// Returns true if x is equipped or inside & usable in what's equipped (currently only applicable to magtractors)
+#define equipped_or_holding(x,source) (source.equipped() == x || (source.equipped()?.useInnerItem && (x in source.equipped())))
```

Comment:
I'm pretty pleased to have gotten this in a define but if anyone knows of a way not to call `equipped` thrice while doing it I'd be interested in hearing.

Reply:
you would have to do one of those multiline do-while-wrapped macros